runtime/shared_heap.c

```diff
@@ -452,19 +452,20 @@ struct pool* caml_pool_of_shared_block(value v)
 
 static intnat pool_sweep(struct caml_heap_state* local, pool** plist,
                          sizeclass sz, int release_to_global_pool) {
-  intnat work = 0;
+  intnat work;
   pool* a = *plist;
   if (!a) return 0;
   *plist = a->next;
 
   {
     value* p = (value*)((char*)a + Bsize_wsize(POOL_SLAB_WOFFSET(sz)));
     value* end = (value*)a + POOL_WSIZE;
+    work = end - p;
```

Comment:
Hoisting `work` out of the loop and into this constant is correct under the assumption that `end - p` is always a multiple of `wh`. Is it the case? I don’t know the internals of the sweeping phase well enough to answer this question.

Reply:
I see this has already been addressed by @dustanddreams and that you responded to it. As you suggest, I am in favor of adding a comment and/or an assertion like ` CAMLassert(p == end);` to make the assertion explicit.